include/jemalloc/internal/tcache_inlines.h

```diff
@@ -33,29 +33,44 @@ tcache_enabled_set(tsd_t *tsd, bool enabled) {
 }
 
 static inline unsigned
-tcache_nhbins_get(tcache_t *tcache) {
+tcache_nbins_get_slow(tcache_slow_t *tcache_slow) {
+	assert(tcache_slow != NULL);
+	assert(tcache_slow->tcache_nbins <= TCACHE_NBINS_MAX);
+	return tcache_slow->tcache_nbins;
+}
+
+static inline size_t
+tcache_max_get_slow(tcache_slow_t *tcache_slow) {
+	assert(tcache_slow != NULL);
+	size_t tcache_max = sz_index2size(tcache_nbins_get_slow(tcache_slow)
+	    - 1);
+	assert(tcache_max <= TCACHE_MAXCLASS_LIMIT);
+	return tcache_max;
+}
+
+static inline unsigned
+tcache_nbins_get(tcache_t *tcache) {
 	assert(tcache != NULL);
-	assert(tcache->tcache_nhbins <= TCACHE_NBINS_MAX);
-	return tcache->tcache_nhbins;
+	return tcache_nbins_get_slow(tcache->tcache_slow);
 }
 
 static inline size_t
 tcache_max_get(tcache_t *tcache) {
 	assert(tcache != NULL);
-	assert(tcache->tcache_max <= TCACHE_MAXCLASS_LIMIT);
-	return tcache->tcache_max;
+	return tcache_max_get_slow(tcache->tcache_slow);
 }
 
 static inline void
-tcache_max_and_nhbins_set(tcache_t *tcache, size_t tcache_max) {
+tcache_max_and_nbins_set(tcache_t *tcache, size_t tcache_max) {
```

Comment:
Let's make this one take tcache_slow as input as well. Probably name it `tcache_max_set` or `tcache_nbins_set` since we now only keep one of them. Similarly for the thread_ variant below.